src/Container/ BooksForm.js

```diff
@@ -11,29 +14,82 @@ const categories = [
   'Sci-Fi',
 ];
 
-const BookForm = () => (
-  <form className="col-8">
-    <div className="mb-3">
-      <label htmlFor="bookTitle" className="form-label">
-        Book title
-      </label>
-      <input type="text" className="form-control" id="bookTitle" />
-    </div>
-    <div className="mb-3">
-      <select className="form-select" aria-label="Default select example">
-        <option defaultValue>Select category</option>
-        {categories.map((category) => (
-          <option value={category} key={category}>
-            {category}
-          </option>
-        ))}
-      </select>
-    </div>
-
-    <button type="submit" className="btn btn-primary">
-      Add book
-    </button>
-  </form>
-);
+const BookForm = () => {
+  const dispatch = useDispatch();
+  const [Book, setBook] = useState({ title: '', category: '' });
+
+  const handleChange = (e) => {
+    let title = '';
+    let category = '';
+    if (e.target.id === 'bookTitle') {
+      title = e.target.value;
+      setBook((state) => ({ ...state, title }));
+    }
+    if (e.target.id === 'bookCategory') {
+      category = e.target.value;
+      setBook((state) => ({ ...state, category }));
+    }
+  };
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    if (Book.title === '') {
+      const alert = document.querySelector('.alert');
+      if (alert) {
+        alert.classList.add('show');
+      }
+      return;
+    }
+
+    dispatch(createAction({ ...Book, id: idGenerator() }));
+    setBook((state) => ({ ...state, title: '', category: '' }));
+  };
+
+  return (
+    <form className="col-8">
```

Comment:
We need to attach `handleSubmit` to the form's `onSubmit` event.

Example: 
`<form {onSubmit={handleSubmit}>`

![image](https://user-images.githubusercontent.com/31897434/126008666-ce411cb1-a44e-402f-b53e-f530416cad92.png)